api/v0/proto/retrievability/types.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.types;
+
+// TODO: oneof different blob versions
+message Blob {
+  // The blob data.
+  bytes data = 1;
+  // The signer of the blob.
+  optional bytes signer = 2;
+  // The blob commitment.
+  bytes commitment = 3;
+}
+
+// Blob with blob proof. Either the blob and the proof are optional, depending
+// on the options of the request.
+message BlobWithProof {
+  // The blob.
+  optional Blob blob = 1;
+  // The proof of blob inclusion or non-inclusion.
+  optional DataRootProof proof = 2;
+  // Index of the first share of the blob in the ODS.
+  uint64 index = 3;
+}
+
+// A single data root proof can span multiple rows.
+message DataRootProof {
+  // The type of the proof.
+  ProofType type = 1;
+  // The proof.
+  Proof proof = 2;
+}
```

Comment:
If all proofs are data root proofs now. How about just get rid of `DataRootProof` message and embed the proof type into `Proof` message? Will use just `Proof` everywhere

Reply:
Fixed in 477ebd59c73042359a888965c28c882dd252c3e0.